File: frontend/planta-smart-4.0/src/infrastructure/mappers/ExpedicaoMapper.ts
import { ExpedicaoRequestDTO } from '@dtos/request/ExpedicaoRequestDTO';
import { ExpedicaoResponseDTO } from '@dtos/response/ExpedicaoResponseDTO';
import { Expedicao } from '@entities/Expedicao';

export const ExpedicaoMapper = {
  mapEntityByResponseDTO(dto: ExpedicaoResponseDTO): Expedicao {
    return {
      id: dto.id,
      posicaoFisica: dto.posicaoFisica,
      ordemDeProducaoAtual: dto.ordemDeProducaoAtual,
    };
  },

  mapEntitiesByResponsesDTOs(dtos: ExpedicaoResponseDTO[]): Expedicao[] {
    const responses: Expedicao[] = [];

    dtos.forEach((dto) => {
      responses.push(this.mapEntityByResponseDTO(dto));
    });

    return responses;
  },

  mapRequestDTOByEntity(entity: Expedicao): ExpedicaoRequestDTO {
    return {
      id: entity.id,
      posicaoFisica: entity.posicaoFisica,
      ordemDeProducao: entity.ordemDeProducaoAtual,
    }
  },

  mapRequestDTOsByEntities(entities: Expedicao[]): ExpedicaoRequestDTO[] {
    const responses: ExpedicaoRequestDTO[] = [];

    entities.forEach((entity) => {
      responses.push(this.mapRequestDTOByEntity(entity));
    })

    return responses;
  },
};

File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/mapper/ExpedicaoMapper.java
package com.smart.appsa.mapper;

import com.smart.appsa.dto.response.ExpedicaoResponseDTO;
import com.smart.appsa.model.Expedicao;

public class ExpedicaoMapper {

    public static ExpedicaoResponseDTO mapDto(Expedicao expedicao) {
        return ExpedicaoResponseDTO.builder()
            .id(expedicao.getId())
            .ordemDeProducaoAtual(expedicao.getOrdemDeProducaoAtual())
            .posicaoFisica(expedicao.getPosicaoFisica())
            .build();
    }

    public static Expedicao mapEntityByResponseDTO(ExpedicaoResponseDTO dto) {
        return Expedicao.builder()
            .id(dto.id())
            .ordemDeProducaoAtual(dto.ordemDeProducaoAtual())
            .posicaoFisica(dto.posicaoFisica())
            .build();
    }
}
